app.py

```diff
@@ -64,23 +62,89 @@ def download_chapter(url):
     """Download a chapter."""
     return requests.get(url).content
 
+def yes_or_no(question):
+    """Prompt user for a y/n answer"""
+    while "the answer is invalid":
+        reply = str(input(question+' (y/n): ')).lower().strip()
+        if reply[:1] == 'y':
+            return True
+        if reply[:1] == 'n':
+            return False
 
 # start the app
 if __name__ == '__main__':
 
-    if len(sys.argv) != 2:
-        print(USAGE_HELP.format(*sys.argv))
-        sys.exit(1)
-
-    book_url = sys.argv[1]
-
-    with open_browser(book_url) as browser:
-        book_id = get_book_id(browser.page_source)
-        playlist = get_playist(browser, book_id)
-
-    for url, fname in playlist:
-        print('Downloading chapter "{}"'.format(fname))
-        with open('{}.mp3'.format(fname), 'wb') as outfile:
-            outfile.write(download_chapter(url))
-
-    print('All done.')
+    #set up command-line arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-o', '--outputDir',
+                        help='Directory the AudioBook will be dowloaded to. Default: <AudioBookName>',
+                        metavar='OutputDirectory',
+                        type=str)
+    parser.add_argument('audioBookURL',
+                        help='a URL of a multi-part audio book',
+                        metavar='URL')
+    parser.add_argument('-w', '--overwrite',
+                        help='overwrite existing directory without asking',
+                        action='store_true',
+                        default=False)
+    parser.add_argument('-c', '--cleanup',
+                        help='delete outputDir in case of errors',
+                        action='store_true',
+                        default=False)
+
+    #parse arguments
+    args = parser.parse_args()
+
+    #default value for output dir is the book name
+    if args.outputDir is None:
+        args.outputDir = args.audioBookURL.split('/')[-1]
```

Comment:
URL/path must be sanitized (potential malicious input).

Reply:
:+1: